libs/util/data-fetcher/src/lib/readers/wfs.ts

```diff
@@ -0,0 +1,117 @@
+import { GeojsonReader, parseGeojson } from './geojson'
+
+export class WfsReader extends BaseReader {
+  endpoint: WfsEndpoint
+  featureTypeName: string
+  version: WfsVersion
+
+  constructor(url: string, wfsEndpoint: WfsEndpoint, featureTypeName: string) {
+    super(url)
+    this.endpoint = wfsEndpoint
+    this.featureTypeName = featureTypeName
+    this.version = this.endpoint.getVersion()
+  }
+
+  get properties(): Promise<PropertyInfo[]> {
+    return this.getData().then((result) => result.properties)
+  }
+
+  get info(): Promise<DatasetInfo> {
+    return this.getData().then(
+      (result) =>
+        ({
+          itemsCount: result.items.length,
+        }) as DatasetInfo
+    )
+  }
+
+  static async createReader(
+    url: string,
+    wfsUrlEndpoint: string,
+    featureTypeName?: string
+  ) {
+    const wfsEndpoint = await new WfsEndpoint(wfsUrlEndpoint).isReady()
+    const featureTypes = wfsEndpoint.getFeatureTypes()
+    const featureType = wfsEndpoint.getFeatureTypeSummary(
+      featureTypes.length === 1 && !featureTypeName
+        ? featureTypes[0].name
+        : featureTypeName
+    )
+    if (!featureType) {
+      throw new Error('wfs.featuretype.notfound')
+    }
+
+    if (wfsEndpoint.supportsStartIndex()) {
+      return new WfsReader(url, wfsEndpoint, featureType.name)
+    } else if (wfsEndpoint.supportsJson(featureType.name)) {
+      return new GeojsonReader(
+        wfsEndpoint.getFeatureUrl(featureType.name, {
+          asJson: true,
+          outputCrs: 'EPSG:4326',
+        })
+      )
+    } else {
+      if (
+        featureType.outputFormats.find((f) =>
+          f.toLowerCase().includes('gml')
+        ) &&
+        (featureType.defaultCrs === 'EPSG:4326' ||
+          featureType.otherCrs?.includes('EPSG:4326'))
+      ) {
+        return new GmlReader(
+          wfsEndpoint.getFeatureUrl(featureType.name, {
+            outputFormat: featureType.outputFormats.find((f) =>
+              f.toLowerCase().includes('gml')
+            ),
+            outputCrs: 'EPSG:4326',
+          }),
+          featureType.name,
+          wfsEndpoint.getVersion()
+        )
+      }
+      throw new Error('wfs.geojsongml.notsupported')
+    }
+  }
+
+  protected getData() {
+    const asJson = this.endpoint.supportsJson(this.featureTypeName)
+    let url = this.endpoint.getFeatureUrl(this.featureTypeName, {
+      ...(this.startIndex !== null && { startIndex: this.startIndex }),
+      ...(this.count !== null && { maxFeatures: this.count }),
+      asJson,
+      outputCrs: 'EPSG:4326',
+      // sortBy: this.sort // TODO: no sort in ogc-client?
+    })
+
+    if (Array.isArray(this.sort) && this.sort.length > 0) {
+      const finalUrl = new URL(url)
+      const sorts = this.sort
+        .map(
+          (fieldSort) => `${fieldSort[1]}+${fieldSort[0] === 'asc' ? 'A' : 'D'}`
+        )
+        .join(',')
+
+      finalUrl.searchParams.append('sortBy', sorts)
+      url = finalUrl.toString()
+    }
```

Comment:
looking good! would be great to have this in ogc-client :)

Reply:
FYI i now set the params directly on the url string as searchParams is encoding the "+"